dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityAttributeStore.java

```diff
@@ -176,6 +183,50 @@ public HibernateTrackedEntityAttributeStore( SessionFactory sessionFactory, Jdbc
             .collect( Collectors.toSet() );
     }
 
+    @Override
+    public Set<TrackedEntityAttribute> getAllSearchableAndUniqueTrackedEntityAttributes()
```

Comment:
I am wondering if the indexing, analyzing, vacuuming etc deserves its own component/class, and it might not fit well in the service + store structure. This is more of a maintenance/admin concern. Maybe something like TrackedEntityAttributeTableManager or similar.

Reply:
@larshelge can you please check again. I think @ameenhere addressed this request 😄

Reply:
Thanks @teleivo 
Yes I have just updated the PR and fixed the broken tests. I have applied all your review comments @larshelge. 
Let me know if its good for merging. 
We need to discuss (including @Bekkalizer ) on how we want this to be backported.